sunpy/net/dataretriever/sources/fermi_gbm.py

```diff
@@ -37,118 +36,28 @@ class GBMClient(GenericClient):
-            det = _check_detector(kwargs['detector'])
-        else:
-            det = 'n5'
-
-        # Check for resolution keyword - either CSPEC or CTIME
-        # Default type is CSPEC
-        if 'resolution' in kwargs:
-            data_type = _check_type(kwargs['resolution'])
-        else:
-            data_type = 'cspec'
-
-        gbm_pattern = ('https://heasarc.gsfc.nasa.gov/FTP/fermi/data/gbm/daily/'
-                       '%Y/%m/%d/current/glg_{data_type}_{det}_%y%m%d_v00.pha')
-        gbm_files = Scraper(gbm_pattern, data_type=data_type, det=det)
-        urls = gbm_files.filelist(timerange)
-
-        return urls
-
-    def _makeimap(self):
-        """
-        Helper function used to hold information about source.
-        """
-        self.map_['source'] = 'FERMI'
-        self.map_['instrument'] = 'GBM'
-        self.map_['physobs'] = 'flux'
-        self.map_['provider'] = 'NASA'
-
-    @classmethod
-    def _can_handle_query(cls, *query):
-        """
-        Answers whether a client can service the query.
-
-        Parameters
-        ----------
-        query : `list`
-            A list of of query objects.
-
-        Returns
-        -------
-        `bool`
-            `True` if this client can service the query, otherwise `False`.
-        """
-        chkattr = ['Time', 'Instrument', 'Detector', 'Resolution']
-        chklist = [x.__class__.__name__ in chkattr for x in query]
-        for x in query:
-            if x.__class__.__name__ == 'Instrument' and x.value.lower() == 'gbm':
-                return all(chklist)
-        return False
+    baseurl = r'https://heasarc.gsfc.nasa.gov/FTP/fermi/data/gbm/daily/%Y/%m/%d/current/glg_(\w){5}_(\w){2}_%y%m%d_v00.pha'
+    pattern = '{}/daily/{year:4d}/{month:2d}/{day:2d}/current/glg_{Resolution:5}_{Detector:2}_{:6d}{}'
 
     @classmethod
     def register_values(cls):
         from sunpy.net import attrs
         adict = {attrs.Instrument: [('GBM', 'Gamma-Ray Burst Monitor on board the Fermi satellite.')],
-                 attrs.Physobs: [('CSPEC', 'counts accumulated every 4.096 seconds in 128 energy channels for each detector.'),
-                                 ('CTIME', 'counts accumulated every 0.256 seconds in 8 energy channels')],
+                 attrs.Physobs: [('flux', 'a measure of the amount of radiation received by an object from a given source.')],
+                 attrs.Source: [('FERMI', 'The Fermi Gamma-ray Space Telescope.')],
+                 attrs.Provider: [('NASA', 'The National Aeronautics and Space Administration.')],
                  attrs.Detector: [
             (f"n{x}", f"GBM Detector short name for the detector NAI_{x:02}") for x in range(12)],
             attrs.Resolution: [
-            ("CSPEC", "CSPEC 128 channel spectra every 4.096 seconds."),
-            ("CTIME", "CTIME provides 8 channel spectra every 0.256 seconds")]
+            ("cspec", "CSPEC 128 channel spectra every 4.096 seconds."),
```

Comment:
I feel like the attrs should be case insensitive, if they are not, we might want to change that.

Reply:
That would have been ideal, but the problem is, the arhive website may contain these attrs in any case, and our pattern matching will fail if we don't follow the website.

If we want to ignore case, that can also be done, but we have to then overwrite `pre_search_hook()` method (which norh does for converting 34 ghz to `tca`). But defining pre-hook for converting `CTIME` to `ctime` or `Ctime` looked like overkill.
So I defined the case the archive follows in register values itself.